liquibase-service/src/main/resources/db/rdb/routines/013-sp_hepatitis_datamart_postprocessing-002.sql

```diff
@@ -0,0 +1,2715 @@
+        FROM dbo.F_PAGE_CASE AS PAGE_CASE WITH(NOLOCK)
+                 INNER JOIN
+             #TMP_F_PAGE_CASE AS TMP_F_PAGE_CASE
+             ON TMP_F_PAGE_CASE.INVESTIGATION_KEY = PAGE_CASE.INVESTIGATION_KEY   ----(my table) since in sas it is dbo
+        ORDER BY D_INVESTIGATION_REPEAT_KEY;
+
+        SELECT inv_repeat.*
+        INTO #D_INV_REPEAT_TEMP_COLS
+        FROM (
+                 SELECT CAST( NULL  AS varchar(1000)) AS VAC_VaccineDoseNum ,
+                        CAST( NULL   AS varchar(2000)) AS VAC_VaccinationDate,
+                        CAST( NULL AS FLOAT) AS PAGE_CASE_UID,
+                        CAST( NULL AS VARCHAR(30)) AS BLOCK_NM,
+                        CAST( NULL AS BIGINT) AS ANSWER_GROUP_SEQ_NBR
+             ) AS temp_inv_repeat_columns
+                 CROSS APPLY
+             ( SELECT D_INVESTIGATION_REPEAT_KEY,
+                      VAC_VaccineDoseNum,
+                      VAC_VaccinationDate,
+                      PAGE_CASE_UID,
+                      BLOCK_NM,
+                      ANSWER_GROUP_SEQ_NBR
+               FROM dbo.D_INVESTIGATION_REPEAT
+             ) AS inv_repeat;
+
+        SELECT DISTINCT
+            F.D_INVESTIGATION_REPEAT_KEY D_INVESTIGATION_REPEAT_KEY1, F.INVESTIGATION_KEY, F.CONDITION_KEY, D.*,
+            RTRIM(LTRIM(VAC_VaccineDoseNum)) AS VAC_VaccineDoseNum1 ,
+            VAC_VaccinationDate as VAC_VaccinationDate1 ,
+            PAGE_CASE_UID PAGE_CASE_UID1, BLOCK_NM BLOCK_NM1, ANSWER_GROUP_SEQ_NBR ANSWER_GROUP_SEQ_NBR1
+        INTO #TMP_D_INVESTIGATION_REPEAT
+        FROM #TMP_F_INVESTIGATION_REPEAT AS F
+                 LEFT JOIN
+             #D_INV_REPEAT_TEMP_COLS AS D WITH(NOLOCK)
+             ON D.D_INVESTIGATION_REPEAT_KEY = F.D_INVESTIGATION_REPEAT_KEY
+        ORDER BY F.INVESTIGATION_KEY;
+
+
+        SELECT @ROWCOUNT_NO = @@ROWCOUNT;
+
+        INSERT INTO [DBO].[JOB_FLOW_LOG]( BATCH_ID, [DATAFLOW_NAME], [PACKAGE_NAME], [STATUS_TYPE], [STEP_NUMBER], [STEP_NAME], [ROW_COUNT] )
+        VALUES( @BATCH_ID, 'HEPATITIS_DATAMART', 'Hepatitis_Case_DATAMART', 'START', @PROC_STEP_NO, @PROC_STEP_NAME, @ROWCOUNT_NO );
+
+        COMMIT TRANSACTION;
+
+
+--------------------------------------------------25. CREATE TABLE TMP_METADATA_TEST---------------------------
+        BEGIN TRANSACTION;
+
+        SET @Proc_Step_name = 'Generating #TMP_METADATA_TEST';
+
+        SET @Proc_Step_no = 25;
+
+        IF OBJECT_ID('#TMP_METADATA_TEST', 'U') IS NOT NULL
+            BEGIN
+                DROP TABLE #TMP_METADATA_TEST;
+
+            END;
+
+        --cte added to assist in filtering latest answers from page case answer table
+        WITH NRT_PCA_FILTERED_CTE AS (
+            SELECT
+                C.condition_key, M.block_nm, M.investigation_form_cd, M.act_uid, M.last_chg_time, max(M.last_chg_time) OVER (PARTITION BY M.investigation_form_cd, act_uid ) max_last_chg_time
+            FROM
+                RDB_MODERN.DBO.NRT_PAGE_CASE_ANSWER AS M WITH(NOLOCK)
```

Comment:
The database name rdb_modern can be removed. We're avoiding specifying the sproc database unless it ODSE or SRTE.